src/libstd/macros.rs

```diff
@@ -444,9 +444,16 @@ pub mod builtin {
     #[macro_export]
     macro_rules! cfg { ($($cfg:tt)*) => ({ /* compiler built-in */ }) }
 
-    /// Parse the current given file as an expression.
-    ///
-    /// This is generally a bad idea, because it's going to behave unhygienically.
+    /// Parse the file provided in the argument as an expression, item, or
+    /// statement according to the context. This file is located relative to the
```

Comment:
I believe the file is can only be parsed as an expression or an item -- c.f. [this `impl`](https://github.com/rust-lang/rust/blob/79fa9eb643cfbb807813afed6f825f6654ee7662/src/libsyntax/ext/source_util.rs#L108).